### PR TITLE
feat: add Swift codes for deque article

### PR DESCRIPTION
**Pull request by nuomi1**
If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by nuomi1** [comment]
```text
result of `deque.swift`

双向队列 deque = [1, 3, 2, 5, 4]
队首元素 peekFirst = 1
队尾元素 peekLast = 4
队首出队元素 pollFirst = 1，队首出队后 deque = [3, 2, 5, 4]
队尾出队元素 pollLast = 4，队尾出队后 deque = [3, 2, 5]
双向队列长度 size = 3
双向队列是否为空 = false
```

File: codes/swift/chapter_stack_and_queue/deque.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * Created Time: 2023-01-14
+ * Author: nuomi1 (nuomi1@qq.com)
+ */
+
+@main
+enum Deque {
+    /* Driver Code */
+    static func main() {
+        /* 初始化双向队列 */
+        // Swift 没有内置的双向队列类，可以把 Array 当作双向队列来使用
```
Comment (krahets) [inline codes/swift/chapter_stack_and_queue/deque.swift:12]: Is the complexity of `poolFirst()` O(n)? If so, please add a comment to mention it.

Comment (krahets) [reply codes/swift/chapter_stack_and_queue/deque.swift:12]: Same for `queue.swift`

**Review by krahets — COMMENTED** [review]
Thanks! Please address the comments.

**Review by krahets — APPROVED** [review]
Thanks!

**Pull request merged**